hw/ip/dcd/data/dcd.hjson

```diff
@@ -0,0 +1,402 @@
+
+	      There are 8 filters to define the potential range(min, max)
+              [11:0]: min value [1:0] RO 0
+              [12]: condition(in the box or out of the box)
+	      [27:16]: max value [17:16] RO 0
+	      each step is 2.148mV
+	      ''',
+        count: "NumAdcFilter",
+        cname: "DCD",
+        hwqe : "true",
+        swaccess: "rw",
+        hwaccess: "hrw",
+        resval: "0",
+        fields: [
+          { bits: "1:0",
+	    name: "min_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "11:2",
+	    name: "min_v",
+	    desc: "10-bit for chn0 filter min value ",
+	  }
+          { bits: "12",
+	    name: "cond",
+	    desc: "1-bit for the condition; 1'b0 means min<=ADC<=max, 1'b1 means ADC>max or ADC<min ",
+	  }
+          { bits: "17:16",
+	    name: "max_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "27:18",
+	    name: "max_v",
+	    desc: "10-bit for chn0 filter max value ",
+	  }
+        ],
+      }
+    }
+    { multireg: {
+        name: "adc_chn1_filter_ctl",
+        desc: '''ADC channel1 filter range
+
+	      There are 8 filters to define the potential range(min, max)
+              [11:0]: min value [1:0] RO 0
+              [12]: condition(in the box or out of the box)
+	      [27:16]: max value [17:16] RO 0
+	      each step is 2.148mV
+	      ''',
+        count: "NumAdcFilter",
+        cname: "DCD",
+        hwqe : "true",
+        swaccess: "rw",
+        hwaccess: "hrw",
+        resval: "0",
+        fields: [
+          { bits: "1:0",
+	    name: "min_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "11:2",
+	    name: "min_v",
+	    desc: "10-bit for chn0 filter min value ",
+	  }
+          { bits: "12",
+	    name: "cond",
+	    desc: "1-bit for the condition; 1'b0 means min<=ADC<=max, 1'b1 means ADC>max or ADC<min ",
+	  }
+          { bits: "17:16",
+	    name: "max_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "27:18",
+	    name: "max_v",
+	    desc: "10-bit for chn0 filter max value ",
+	  }
+        ],
+      }
+    }
+    { multireg: {
+        name: "adc_chn_val",
+        desc: "ADC value sampled on channel",
+        count: "NumAdcChannel",
+        cname: "DCD",
+        hwqe : "true",
+        swaccess: "ro",
+        hwaccess: "hrw",
+        resval: "0",
+        fields: [
+          { bits: "1:0",
+	    name: "adc_chn_value_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "11:2",
+	    name: "adc_chn_value",
+	    desc: "Latest ADC value sampled on channel. each step is 2.148mV",
+	  }
+          { bits: "17:16",
+	    name: "adc_chn_value_intr_ext",
+	    desc: "2-bit extension; RO 0",
+	  }
+          { bits: "27:18",
+	    name: "adc_chn_value_intr",
+	    desc: "ADC value sampled on channel when the interrupt is raised(debug cable is attached or disconnected)each step is 2.148mV",
+	  }
+        ],
+      }
+    }
+    { name: "adc_wakeup_ctl",
```

Comment:
you could use multireg here too if you want.

Reply:
It's only 8-bit. I will pass the multireg optimization.